_summaries/katharopoulos2020lineartransformer_2.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: summary
+title: Summary
+paper: {{katharopoulos2020lineartransformer}}
+author:  sritank
+score:  10
+---
+
+## What is the core idea?
+A new linear transformer model is introduced that uses a kernel based self-attention method that reduces the memory footprint of the model. They also linearize the masking of the attention computation making it run in linear time.
+
+## How is it realized (technically)?
+
+They change the attention mechanism from the traditional softmax function to a dot product feature map. In a traditional transformer, given the queries Q, keys K and values V, the self attention function A_l can be calculated as:
+
+![self attention function](./katharopoulos2020lineartransformer_2a.png)
+
+where,
+
+![softmax kernel](./katharopoulos2020lineartransformer_2b.png)
+
+Instead of a softmax similarity function, the authors propose using a kernel function of a feature representation \phi(x). So the new self attention function is:
+
+![linear self attention function](./katharopoulos2020lineartransformer_2c.png)
+
+The above attention function has computation cost of O(N) where N is the sequence length. It's memory footprint is also lower as the terms can be reused for every query hence we don't need to store the full attention matrix making it's memory complexity O(N). The authors state that for small sequences the below feature map, phi(x), was used and shown to perform on par to the full traditional transformer.
+
+They further change the masking function to span until the i-th position of the query instead of the whole sequence (i.e. all N inputs). They called this ``causal'' masking. They further show how their formulation of a transformer with causal masking can be understood as an RNN.
+
+![causal masking](./katharopoulos2020lineartransformer_2d.png)
+
+
```

Comment:
Just for some more context, you could also specify which part of the self-attention computation is being stored as a hidden state in the RNN representation of the transformer layer - phi (K_j) V^T_j.